[dynamo, guards] apply functools.cached_property to Source.name

williamwen42 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #168886
* #168386
* #168203
* __->__ #168131

Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 25s -> 16s on a local tlparse. On the guard build benchmark, time went from 135.81s -> 84.66s

However, there are a lot of changes to `source.name()` callsites. This could technically be avoided by writing our own `cached_property` decorator that requires the function call.

cc @ezyang @EikanWang @jgong5 @wenzhe-nrv @voznesenskym @penguinwu @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @jiayisunx @kadeng @chauhang @amjames @Lucaskabela @jataylo @chenyang78

williamwen42 committed:
[dynamo, guards] apply functools.cached_property to Source.name

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] apply functools.cached_property to Source.name"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 25s -> 16s locally.

However, there are a lot of changes to `source.name()` callsites. This could technically be avoided by writing our own `cached_property` decorator that requires the function call.

cc ezyang EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] apply functools.cached_property to Source.name"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 25s -> 16s locally.

However, there are a lot of changes to `source.name()` callsites. This could technically be avoided by writing our own `cached_property` decorator that requires the function call.

cc ezyang EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] apply functools.cached_property to Source.name"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 25s -> 16s locally.

However, there are a lot of changes to `source.name()` callsites. This could technically be avoided by writing our own `cached_property` decorator that requires the function call.

cc ezyang EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] apply functools.cached_property to Source.name"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 25s -> 16s locally.

However, there are a lot of changes to `source.name()` callsites. This could technically be avoided by writing our own `cached_property` decorator that requires the function call.

cc ezyang EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] apply functools.cached_property to Source.name"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 25s -> 16s locally.

However, there are a lot of changes to `source.name()` callsites. This could technically be avoided by writing our own `cached_property` decorator that requires the function call.

cc ezyang EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] apply functools.cached_property to Source.name"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 25s -> 16s locally.

However, there are a lot of changes to `source.name()` callsites. This could technically be avoided by writing our own `cached_property` decorator that requires the function call.

cc ezyang EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] apply functools.cached_property to Source.name"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 25s -> 16s locally.

However, there are a lot of changes to `source.name()` callsites. This could technically be avoided by writing our own `cached_property` decorator that requires the function call.

cc ezyang EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] apply functools.cached_property to Source.name"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 25s -> 16s on a local tlparse. On the guard build benchmark, time went from 135.81s -> 84.66s

However, there are a lot of changes to `source.name()` callsites. This could technically be avoided by writing our own `cached_property` decorator that requires the function call.

cc ezyang EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update

[ghstack-poisoned]

anijain2305 reviewed (approved):
Thank you! I am pretty sure this was very tedious to do.

aorenste commented on torch/_dynamo/source.py:
```diff
@@ -162,6 +163,7 @@ def reconstruct(self, codegen: "PyCodegen") -> None:
         return GuardSource.TEMP_LOCAL
 
+    @functools.cached_property
     def name(self) -> str:
```
You can use `@property` instead, can't you?

williamwen42 committed:
Update

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] apply functools.cached_property to Source.name"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 25s -> 16s on a local tlparse. On the guard build benchmark, time went from 135.81s -> 84.66s

However, there are a lot of changes to `source.name()` callsites. This could technically be avoided by writing our own `cached_property` decorator that requires the function call.

cc ezyang EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] apply functools.cached_property to Source.name"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 25s -> 16s on a local tlparse. On the guard build benchmark, time went from 135.81s -> 84.66s

However, there are a lot of changes to `source.name()` callsites. This could technically be avoided by writing our own `cached_property` decorator that requires the function call.

cc ezyang EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]

williamwen42 committed:
Update on "[dynamo, guards] apply functools.cached_property to Source.name"


Partial fix for https://github.com/pytorch/pytorch/issues/168118. Decreases guard build time from 25s -> 16s on a local tlparse. On the guard build benchmark, time went from 135.81s -> 84.66s

However, there are a lot of changes to `source.name()` callsites. This could technically be avoided by writing our own `cached_property` decorator that requires the function call.

cc ezyang EikanWang jgong5 wenzhe-nrv voznesenskym penguinwu Guobing-Chen XiaobingSuper zhuhaozhe blzheng jiayisunx chenyang78 kadeng chauhang amjames Lucaskabela

[ghstack-poisoned]
